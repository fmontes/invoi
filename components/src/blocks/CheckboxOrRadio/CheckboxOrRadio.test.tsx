import * as React from 'react';
import { CheckboxOrRadio } from './index';
import { shallow } from 'enzyme';

describe('Checkbox', () => {
    test('Checkbox should render', () => {
        const component = shallow(<CheckboxOrRadio type="checkbox" />);
        expect(component).toMatchSnapshot();
    });

    test('Checkbox should render disaled', () => {
        const component = shallow(<CheckboxOrRadio disabled type="checkbox" />);
        expect(component).toMatchSnapshot();
    });

    test('Checkbox should render error', () => {
        const component = shallow(<CheckboxOrRadio error type="checkbox" />);
        expect(component).toMatchSnapshot();
    });
});

describe('Radio', () => {
    test('Radio should render', () => {
        const component = shallow(<CheckboxOrRadio type="radio" />);
        expect(component).toMatchSnapshot();
    });

    test('Radio should render disaled', () => {
        const component = shallow(<CheckboxOrRadio disabled type="radio" />);
        expect(component).toMatchSnapshot();
    });

    test('Radio should render error', () => {
        const component = shallow(<CheckboxOrRadio error type="radio" />);
        expect(component).toMatchSnapshot();
    });
});
