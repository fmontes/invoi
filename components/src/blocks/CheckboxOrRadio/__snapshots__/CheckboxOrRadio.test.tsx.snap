// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checkbox should render 1`] = `
<Fragment>
  <styled.input
    error={false}
    theme={
      Object {
        "animation": Object {
          "speed": "100ms",
        },
        "base": Object {
          "radius": Object {
            "normal": "0.5rem",
            "tiny": "0.25rem",
          },
        },
        "colors": Object {
          "background": "#FDFCF9",
          "black": "#202641",
          "disabled": "#D7D0CB",
          "error": "#832400",
          "gray": "#5B6078",
          "neutral": "#CBC1BB",
          "primary": "#3951C5",
          "secondary": "#F0A78B",
          "success": "#3951C5",
          "textBody": "#202641",
          "textSub": "#5B6078",
          "white": "#fff",
        },
        "components": Object {
          "buttons": Object {
            "normal": Object {
              "height": "2.75rem",
            },
            "tiny": Object {
              "height": "1.5rem",
            },
          },
        },
        "font": Object {
          "family": "'Work Sans', sans-serif;",
          "size": Object {
            "body": "1rem",
            "h1": "2rem",
            "h2": "1.5rem",
            "hint": "0.8125rem",
            "tiny": "0.6875rem",
          },
        },
        "spaces": Object {
          "half": "0.25rem",
          "one": "0.5rem",
          "three": "2rem",
          "two": "1rem",
        },
      }
    }
    type="checkbox"
  />
   
  <span />
</Fragment>
`;

exports[`Checkbox Checkbox should render disaled 1`] = `
<Fragment>
  <styled.input
    disabled={true}
    error={false}
    theme={
      Object {
        "animation": Object {
          "speed": "100ms",
        },
        "base": Object {
          "radius": Object {
            "normal": "0.5rem",
            "tiny": "0.25rem",
          },
        },
        "colors": Object {
          "background": "#FDFCF9",
          "black": "#202641",
          "disabled": "#D7D0CB",
          "error": "#832400",
          "gray": "#5B6078",
          "neutral": "#CBC1BB",
          "primary": "#3951C5",
          "secondary": "#F0A78B",
          "success": "#3951C5",
          "textBody": "#202641",
          "textSub": "#5B6078",
          "white": "#fff",
        },
        "components": Object {
          "buttons": Object {
            "normal": Object {
              "height": "2.75rem",
            },
            "tiny": Object {
              "height": "1.5rem",
            },
          },
        },
        "font": Object {
          "family": "'Work Sans', sans-serif;",
          "size": Object {
            "body": "1rem",
            "h1": "2rem",
            "h2": "1.5rem",
            "hint": "0.8125rem",
            "tiny": "0.6875rem",
          },
        },
        "spaces": Object {
          "half": "0.25rem",
          "one": "0.5rem",
          "three": "2rem",
          "two": "1rem",
        },
      }
    }
    type="checkbox"
  />
   
  <span />
</Fragment>
`;

exports[`Checkbox Checkbox should render error 1`] = `
<Fragment>
  <styled.input
    error={true}
    theme={
      Object {
        "animation": Object {
          "speed": "100ms",
        },
        "base": Object {
          "radius": Object {
            "normal": "0.5rem",
            "tiny": "0.25rem",
          },
        },
        "colors": Object {
          "background": "#FDFCF9",
          "black": "#202641",
          "disabled": "#D7D0CB",
          "error": "#832400",
          "gray": "#5B6078",
          "neutral": "#CBC1BB",
          "primary": "#3951C5",
          "secondary": "#F0A78B",
          "success": "#3951C5",
          "textBody": "#202641",
          "textSub": "#5B6078",
          "white": "#fff",
        },
        "components": Object {
          "buttons": Object {
            "normal": Object {
              "height": "2.75rem",
            },
            "tiny": Object {
              "height": "1.5rem",
            },
          },
        },
        "font": Object {
          "family": "'Work Sans', sans-serif;",
          "size": Object {
            "body": "1rem",
            "h1": "2rem",
            "h2": "1.5rem",
            "hint": "0.8125rem",
            "tiny": "0.6875rem",
          },
        },
        "spaces": Object {
          "half": "0.25rem",
          "one": "0.5rem",
          "three": "2rem",
          "two": "1rem",
        },
      }
    }
    type="checkbox"
  />
   
  <span />
</Fragment>
`;

exports[`Radio Radio should render 1`] = `
<Fragment>
  <styled.input
    error={false}
    theme={
      Object {
        "animation": Object {
          "speed": "100ms",
        },
        "base": Object {
          "radius": Object {
            "normal": "0.5rem",
            "tiny": "0.25rem",
          },
        },
        "colors": Object {
          "background": "#FDFCF9",
          "black": "#202641",
          "disabled": "#D7D0CB",
          "error": "#832400",
          "gray": "#5B6078",
          "neutral": "#CBC1BB",
          "primary": "#3951C5",
          "secondary": "#F0A78B",
          "success": "#3951C5",
          "textBody": "#202641",
          "textSub": "#5B6078",
          "white": "#fff",
        },
        "components": Object {
          "buttons": Object {
            "normal": Object {
              "height": "2.75rem",
            },
            "tiny": Object {
              "height": "1.5rem",
            },
          },
        },
        "font": Object {
          "family": "'Work Sans', sans-serif;",
          "size": Object {
            "body": "1rem",
            "h1": "2rem",
            "h2": "1.5rem",
            "hint": "0.8125rem",
            "tiny": "0.6875rem",
          },
        },
        "spaces": Object {
          "half": "0.25rem",
          "one": "0.5rem",
          "three": "2rem",
          "two": "1rem",
        },
      }
    }
    type="radio"
  />
   
  <span />
</Fragment>
`;

exports[`Radio Radio should render disaled 1`] = `
<Fragment>
  <styled.input
    disabled={true}
    error={false}
    theme={
      Object {
        "animation": Object {
          "speed": "100ms",
        },
        "base": Object {
          "radius": Object {
            "normal": "0.5rem",
            "tiny": "0.25rem",
          },
        },
        "colors": Object {
          "background": "#FDFCF9",
          "black": "#202641",
          "disabled": "#D7D0CB",
          "error": "#832400",
          "gray": "#5B6078",
          "neutral": "#CBC1BB",
          "primary": "#3951C5",
          "secondary": "#F0A78B",
          "success": "#3951C5",
          "textBody": "#202641",
          "textSub": "#5B6078",
          "white": "#fff",
        },
        "components": Object {
          "buttons": Object {
            "normal": Object {
              "height": "2.75rem",
            },
            "tiny": Object {
              "height": "1.5rem",
            },
          },
        },
        "font": Object {
          "family": "'Work Sans', sans-serif;",
          "size": Object {
            "body": "1rem",
            "h1": "2rem",
            "h2": "1.5rem",
            "hint": "0.8125rem",
            "tiny": "0.6875rem",
          },
        },
        "spaces": Object {
          "half": "0.25rem",
          "one": "0.5rem",
          "three": "2rem",
          "two": "1rem",
        },
      }
    }
    type="radio"
  />
   
  <span />
</Fragment>
`;

exports[`Radio Radio should render error 1`] = `
<Fragment>
  <styled.input
    error={true}
    theme={
      Object {
        "animation": Object {
          "speed": "100ms",
        },
        "base": Object {
          "radius": Object {
            "normal": "0.5rem",
            "tiny": "0.25rem",
          },
        },
        "colors": Object {
          "background": "#FDFCF9",
          "black": "#202641",
          "disabled": "#D7D0CB",
          "error": "#832400",
          "gray": "#5B6078",
          "neutral": "#CBC1BB",
          "primary": "#3951C5",
          "secondary": "#F0A78B",
          "success": "#3951C5",
          "textBody": "#202641",
          "textSub": "#5B6078",
          "white": "#fff",
        },
        "components": Object {
          "buttons": Object {
            "normal": Object {
              "height": "2.75rem",
            },
            "tiny": Object {
              "height": "1.5rem",
            },
          },
        },
        "font": Object {
          "family": "'Work Sans', sans-serif;",
          "size": Object {
            "body": "1rem",
            "h1": "2rem",
            "h2": "1.5rem",
            "hint": "0.8125rem",
            "tiny": "0.6875rem",
          },
        },
        "spaces": Object {
          "half": "0.25rem",
          "one": "0.5rem",
          "three": "2rem",
          "two": "1rem",
        },
      }
    }
    type="radio"
  />
   
  <span />
</Fragment>
`;
